components/identity-mgt/org.wso2.carbon.identity.mgt/src/main/java/org/wso2/carbon/identity/mgt/IdentityMgtEventListener.java

```diff
@@ -814,8 +814,7 @@ public boolean doPreSetUserClaimValues(String userName, Map<String, String> clai
                 try {
                     identityDataStore.store(identityDTO, userStoreManager);
                 } catch (IdentityException e) {
-                    throw new UserStoreException(
-                            "Error while saving user store data for user : " + userName, e);
+                    throw new UserStoreException(e.getMessage(), e);
```

Comment:
We need to throw an informative Axisfault wrapping the exact cause. Therefore, we're catching the exact exception from identityDataStore.store(identityDTO, userStoreManager) and wrapping the message so that the admin service gives an informative fault message.